k2/csrc/fsa.h

```diff
@@ -50,13 +50,20 @@ struct Arc {
            std::tie(src_state, other.dest_state, other.label);
   }
 
+  bool operator!=(const Arc& other) const { return !(*this == other); }
+
   bool operator<(const Arc &other) const {
     // compares `label` first, then `dest_state`
     return std::tie(label, dest_state) <
            std::tie(other.label, other.dest_state);
   }
 };
 
+std::ostream &operator<<(std::ostream &os, const Arc &arc);
+
+template <class T>
+void hash_combine(std::size_t *seed, const T &v);
```

Comment:
why do we need to declare `hash_combine` here again? I thought I put one inline version in `util.h`

Reply:
Consider what will happen while compiling `util.cc`:

(1) It includes `k2/csrc/util.h`, which defines a header guard to avoid including itself multiple times
(2) `k2/csrc/util.h` includes `k2/csrc/fsa.h`
(3) `k2/csrc/fsa.h` includes `k2/csrc/util.h`, which is an no-op because of
the header guard of `k2/csrc/util.h`, which means `hash_combine` is not visible
in `k2/csrc/fsa.h`

Reply:
There are circular dependencies between `k2/csrc/util.h` and `k2/csrc/fsa.h`,
which indicates a bad design.

Reply:
It may be a history issue to include `fsa.h` in `util.h`. Actually we can remove it from `util.h` as it seems there's no dependency on FSA in `util.h`

Reply:
fixed.